### PR TITLE
Expand Launchable into yjit-macos and yjit-ubuntu

### PR DESCRIPTION
**Pull request by ono-max**
Follow-up to https://github.com/ruby/ruby/pull/10086

Since the integration of Launchable into the macOS and Ubuntu workflow has been completed in the previous PR, this PR extends the integration to the yjit-macos and yjit-ubuntu workflow as well.

### CODE REVIEW

**Comment by nobu** [comment]
```diff
diff --git a/.github/workflows/yjit-ubuntu.yml b/.github/workflows/yjit-ubuntu.yml
index cb11061a155..fddf7b402d0 100644
--- a/.github/workflows/yjit-ubuntu.yml
+++ b/.github/workflows/yjit-ubuntu.yml
@@ -166,7 +166,10 @@ jobs:
         run: ./miniruby --yjit -v | grep "+YJIT"
 
       - name: make ${{ matrix.test_task }}
-        run: make -s -j ${{ matrix.test_task }} RUN_OPTS="$RUN_OPTS" MSPECOPT=--debug YJIT_BENCH_OPTS="$YJIT_BENCH_OPTS" YJIT_BINDGEN_DIFF_OPTS="$YJIT_BINDGEN_DIFF_OPTS"
+        run: >-
+          make -s -j ${{ matrix.test_task }} ${TESTS:+TESTS="$TESTS"}
+          RUN_OPTS="$RUN_OPTS" MSPECOPT=--debug
+          YJIT_BENCH_OPTS="$YJIT_BENCH_OPTS" YJIT_BINDGEN_DIFF_OPTS="$YJIT_BINDGEN_DIFF_OPTS"
         timeout-minutes: 60
         env:
           RUBY_TESTOPTS: '-q --tty=no'
```

File: .github/workflows/yjit-ubuntu.yml [hunk 1]
```diff
@@ -165,6 +167,15 @@ jobs:
...
+        uses: ./.github/actions/launchable/setup
+        with:
+          os: ubuntu-20.04
+          test-opts: ${{ matrix.configure }}
+          launchable-token: ${{ secrets.LAUNCHABLE_TOKEN }}
+          builddir: build
+          srcdir: src
+
       - name: make ${{ matrix.test_task }}
         run: make -s -j ${{ matrix.test_task }} RUN_OPTS="$RUN_OPTS" MSPECOPT=--debug YJIT_BENCH_OPTS="$YJIT_BENCH_OPTS" YJIT_BINDGEN_DIFF_OPTS="$YJIT_BINDGEN_DIFF_OPTS"
```
Comment (nobu) [inline .github/workflows/yjit-ubuntu.yml:180]:
```suggestion
        run: >-
          make -s -j ${{ matrix.test_task }} ${TESTS:+TESTS="$TESTS"}
          RUN_OPTS="$RUN_OPTS" MSPECOPT=--debug
          YJIT_BENCH_OPTS="$YJIT_BENCH_OPTS" YJIT_BINDGEN_DIFF_OPTS="$YJIT_BINDGEN_DIFF_OPTS"
```

**Review by maximecb — COMMENTED** [review]
Looks correct, but what is launchable, why this change? 😅

**Comment by ono-max** [comment]
@maximecb 

Here is a summary about your question.

> what is launchable

Launchable is a service designed to address issues related to testing. By integrating Launchable, we expect to receive the following benefits:

* **Rich UI for viewing stderr and stdout in failed tests:**
Reviewing test failure logs in GitHub Actions can be challenging. Launchable provides a user-friendly interface that facilitates the examination of stderr and stdout related to test failures. [Here](https://app.launchableinc.com/organizations/ruby/workspaces/ruby/data/test-sessions/2708884) is an example.

* **Test reports in GitHub comments:**
Identifying which CI job failed in the GitHub UI can be difficult. Launchable tackles this issue by generating a comment that summarizes test results directly in the GitHub pull request, effectively serving as a dashboard. [Here](https://github.com/ruby/ruby/pull/10215#issuecomment-1987762576) is an example.

* **Insight into flakey tests:**
In the ruby/ruby project, there are some flakey tests. Launchable offers an "Unhealthy Tests" page that analyzes these flakey tests. For instance, the page displays a flakiness score indicating the extent of the test's flakiness. [Here](https://app.launchableinc.com/organizations/ruby/workspaces/ruby/insights/unhealthy-tests?year-week=2024-W08) is an example. By using this feature, @mame san identified a bug ([here](https://bugs.ruby-lang.org/issues/20314)).

For discussion on using Launchable, please refer to [this link](https://bugs.ruby-lang.org/issues/20254).

> why this change

To utilize Launchable, we need to upload test results to their platform. Currently, the test results of `make test-all` have been uploaded to Launchable for [Ubuntu](https://github.com/ruby/ruby/blob/master/.github/workflows/ubuntu.yml) and [MacOS](https://github.com/ruby/ruby/blob/master/.github/workflows/macos.yml).

To extend this functionality to include the test results from `yjit-macos` and `yjit-ubuntu`, I have created this pull request. The composite action [here](https://github.com/ruby/ruby/blob/master/.github/actions/launchable/setup/action.yml) will enable Launchable.

**Review by hsbt — APPROVED** [review]

**Pull request merged**